### PR TITLE
Task Based Async Refactor

### PR DESCRIPTION
**Pull request by MelbourneDeveloper**
- Convert all synchronous code to Task based async
- Convert Unit tests to work using await keyword

**Breaking changes**
- All async calls must be awaited
- All async calls are suffixed with Async

_Note: this is the first step to getting the code to work across all platforms: especially UWP._

### CODE REVIEW

File: LedgerWallet/Transports/ILedgerTransport.cs [hunk 1]
```diff
@@ -8,6 +8,6 @@ namespace LedgerWallet.Transports
 {
 	public interface ILedgerTransport
 	{
-		Task<byte[][]> Exchange(byte[][] apdus);
+		Task<byte[][]> ExchangeAsync(byte[][] apdus);
```
Comment (NicolasDorier) [inline LedgerWallet/Transports/ILedgerTransport.cs]: Can you remove this change? it is breaking existing code for no reason.

File: LedgerWallet.Tests/U2FTests.cs [hunk 2]
```diff
@@ -12,39 +12,39 @@
...
+    [Trait("U2F", "U2F")]
+    public class U2FTests
+    {
+        [Fact]
+        [Trait("Manual", "Manual")]
+        public async Task CanEnrollAndAuthenticate()
+        {
+            var appId = new AppId(Encoders.Hex.DecodeData("d2e42c173c857991d5e1b6c81f3e07cbb9d5f57431fe41997c9445c14ce61ec4"));
+            var challenge = Encoders.Hex.DecodeData("e6425678fbd7d3d8e311fbfb1db8d26c37cf9f16ac81c95848998a76ce3d3768");
+            var u2f = (U2FClient)await NanoSTests.GetLedgerAsync(NanoSTests.LedgerType.U2F);
```
Comment (NicolasDorier) [inline LedgerWallet.Tests/U2FTests.cs]: keep tabs

File: LedgerWallet/Transports/ILedgerTransport.cs (hunk 1 above)
Comment (MelbourneDeveloper) [reply LedgerWallet/Transports/ILedgerTransport.cs]: No problem.

Are you sure? It's a Microsoft convention to suffix all task based method with Async. When I look at the code, and I see a method without Async, I assume it's synchronous.

File: LedgerWallet.Tests/U2FTests.cs (hunk 2 above)
Comment (MelbourneDeveloper) [reply LedgerWallet.Tests/U2FTests.cs]: I will revert my formatting changes here, but the original has an extra tab. I did keep the tabs. The problem is that the original is not formatted the way Visual Studio formats tabs. It's different to the other classes.

Comment (MelbourneDeveloper) [reply LedgerWallet.Tests/U2FTests.cs]: Oops. This was a visual studio bug. I did a format for tabs, but it kept the spaces because nothing had changed.

**Comment by MelbourneDeveloper** [comment]
All done.

**Comment by NicolasDorier** [comment]
super nice

**Pull request merged**